packages/core/src/StateMachine.ts

```diff
@@ -450,6 +450,17 @@ export class StateMachine<
       }
     });
 
+    const context = JSON.parse(JSON.stringify(state.context), (_, value) => {
+      // if value is an object with $$type property, it is a reference to an actor
+      if (typeof value === 'object' && value && value.$$type === 'ActorRef') {
+        return children[value.id];
+      }
+
+      return value;
+    });
+
+    restoredState.context = context;
```

Comment:
nit-level but I think that it would just be better to create this rehydrated context first and create `State` right away with it instead of re-assigning it after the fact